internal/runners/branch/tree.go

```diff
@@ -0,0 +1,192 @@
+package branch
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/go-openapi/strfmt"
+)
+
+type branchNode struct {
+	Label string
+	ID    string
+}
+
+type tree map[branchNode]tree
+
+type BranchTree struct {
+	tree                  tree
+	branches              mono_models.Branches
+	rootBranches          mono_models.Branches
+	localBranch           string
+	branchFormatting      string
+	localBranchFormatting string
+}
+
+const (
+	edgeLink string = "│  "
+	edgeMid  string = "├─"
+	edgeEnd  string = "└─"
+)
+
+func NewBranchTree() *BranchTree {
+	return &BranchTree{tree: make(tree)}
+}
+
+func (bt *BranchTree) BuildFromBranches(branches mono_models.Branches) {
+	bt.rootBranches = getRootBranches(branches)
+	for _, branch := range bt.rootBranches {
+		bt.tree[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
```

Comment:
I don't think you need to be storing `rootBranches` and `tree`, just pass them around instead. With the current implementation the state of your `struct` may at times be practically corrupted unless you just "know" which method to call next. It's best if the struct itself is never in a state where it will implicitly do something unexpected.